src/main.ts

```diff
@@ -22,49 +24,95 @@ interface ListFunctionsOptions {
   basePath?: string
   config?: Config
   featureFlags?: FeatureFlags
+  parseISC?: boolean
+}
+
+interface AugmentedFunctionSource extends FunctionSource {
+  inSourceConfig?: ISCValues
+}
+
+const augmentWithISC = async (func: FunctionSource): Promise<AugmentedFunctionSource> => {
+  if (func.runtime.name !== 'js') {
+    return func
+  }
+
+  const inSourceConfig = await findISCDeclarationsInPath(func.mainFile)
+  return { ...func, inSourceConfig }
 }
 
 // List all Netlify Functions main entry files for a specific directory
 const listFunctions = async function (
   relativeSrcFolders: string | string[],
-  { featureFlags: inputFeatureFlags }: { featureFlags?: FeatureFlags } = {},
+  {
+    featureFlags: inputFeatureFlags,
+    config,
+    parseISC = false,
+  }: { featureFlags?: FeatureFlags; config?: Config; parseISC?: boolean } = {},
 ) {
   const featureFlags = getFlags(inputFeatureFlags)
   const srcFolders = resolveFunctionsDirectories(relativeSrcFolders)
   const paths = await listFunctionsDirectories(srcFolders)
-  const functions = await getFunctionsFromPaths(paths, { featureFlags })
-  const listedFunctions = [...functions.values()].map(getListedFunction)
-  return listedFunctions
+  const functionsMap = await getFunctionsFromPaths(paths, { featureFlags, config })
+  const functions = [...functionsMap.values()]
+  const augmentedFunctions = parseISC ? await Promise.all(functions.map(augmentWithISC)) : functions
+  return augmentedFunctions.map(getListedFunction)
+}
+
+// List one Netlify Functions main entry file for a specific directory
```

Comment:
I think this is a more accurate description?

```suggestion
// Finds a function at a specific path.
```

Reply:
I like it!